AzureMonitorAgent/agent.py

```diff
@@ -310,8 +310,25 @@ def install():
     if public_settings is not None and public_settings.get("GCS_AUTO_CONFIG") == "true":
         hutil_log_info("Detecting Auto-Config mode.")
         return 0, ""
-    elif protected_settings is None or len(protected_settings) == 0:
+    elif (protected_settings is None or len(protected_settings) == 0) or (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application"):
```

Comment:
Minor: I understand that it's "OR" clause, but I got a bit confused that on entry into the "ELIF" we check "protected_settings is None", but then on line 326 we check for "protected_settings is not None". Altough, technically it seems to work out just fine, it's a bit consusing in reading - is there a way to restructure IF/ELIF for readability?

Reply:
The commonality in the elif is enabling MCS